torch_xla/csrc/runtime/stablehlo_composite_helper.cc

```diff
@@ -190,35 +190,74 @@ class BuildStableHLOCompositePass : public mlir::OperationPass<mlir::ModuleOp> {
     return metadata;
   }
 
+  mlir::FailureOr<mlir::Attribute> BuildAttrFromJson(mlir::OpBuilder& builder,
+                                                     mlir::Operation* op,
+                                                     const json& json_value) {
+    switch (json_value.type()) {
+      case json::value_t::number_integer:
+      case json::value_t::number_unsigned:
+        return builder.getI64IntegerAttr(json_value.template get<int64_t>());
+      case json::value_t::number_float:
+        return builder.getF32FloatAttr(json_value.template get<float>());
+      case json::value_t::boolean:
+        return builder.getBoolAttr(json_value.template get<bool>());
+      case json::value_t::string:
+        return builder.getStringAttr(json_value.template get<std::string>());
+      case json::value_t::array: {
+        if (json_value.empty()) {
+          return builder.getArrayAttr({});
+        }
+        auto get_json_type = [](const json& j) {
+          auto ty = j.type();
+          if (ty == json::value_t::number_unsigned) {
+            return json::value_t::number_integer;
+          }
+          return ty;
+        };
+
+        auto head_type = get_json_type(json_value[0]);
+        bool is_homogeneous = llvm::all_of(json_value, [&](auto& el) {
+          return get_json_type(el) == head_type;
+        });
+        if (!is_homogeneous) {
+          return op->emitError()
+                 << "invalid JSON to MLIR, arrays must be homogeneous";
+        }
+
+        switch (head_type) {
+          case json::value_t::number_integer:
+            return builder.getDenseI64ArrayAttr(
```

Comment:
I'm just remembering that we don't have general serialization support for DenseArray types. Would recommend using 1-D DenseIntOrFpElements for now:

```
// I64
builder.getI64TensorAttr(values);

//float
return DenseFPElementsAttr::get(RankedTensorType::get(static_cast<float>(values.size()), builder.getF32Type()), values);

//bool
return DenseIntElementsAttr::get(RankedTensorType::get(static_cast<int64_t>(values.size()), builder.getI1Type()), values);
```

where `values` is the small vector cast of JSON you have here.

Reply:
Confirm: does this look good to you?
```
module @IrToHlo.11 attributes {mhlo.cross_program_prefetches = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false} {
  func.func @main(%arg0: tensor<5x5xf32>, %arg1: tensor<5x5xf32>) -> tensor<5x5xf32> {
    %0 = stablehlo.composite "test.add" %arg1, %arg0 {composite_attributes = {bool_arr = dense<[true, false]> : tensor<2xi1>, float_arr = dense<[1.000000e+00, 1.100000e+00, 1.200000e+00]> : tensor<3xf32>, int_arr = dense<[1, 2, 3]> : tensor<3xi64>}, decomposition = @test.add.impl} : (tensor<5x5xf32>, tensor<5x5xf32>) -> tensor<5x5xf32>
    return %0 : tensor<5x5xf32>
  }
  func.func private @test.add.impl(%arg0: tensor<5x5xf32>, %arg1: tensor<5x5xf32>) -> tensor<5x5xf32> {
    %0 = stablehlo.add %arg0, %arg1 : tensor<5x5xf32>
    return %0 : tensor<5x5xf32>
  }
}
```